### ISSUE TITLE
Support compiling with GCC 8 on Ubuntu 18.04

### ISSUE THREAD

**Issue by xkszltl** [ccache/ccache#1186]
### How to reproduce ###

```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=gcc-8 -DCMAKE_CXX_COMPILER=g++-8 -DCMAKE_C_COMPILER_LAUNCHER=/usr/bin/ccache -DCMAKE_CXX_COMPILER_LAUNCHER=/usr/bin/ccache '-DCMAKE_C
_FLAGS=-fdebug-prefix-map='\''/tmp/scratch'\''='\''/usr/local/src'\'' -g' '-DCMAKE_CXX_FLAGS=-fdebug-prefix-map='\''/tmp/scratch'\''='\''/usr/local/src'\'' -g' -DCMAKE_INSTALL_PREFIX=/tmp/scratch/cc
ache/install.vfiLZXpBTA/root/usr/local -DENABLE_IPO=ON -DENABLE_TESTING=ON -DHIREDIS_FROM_INTERNET=OFF -DREDIS_STORAGE_BACKEND=OFF -DZSTD_FROM_INTERNET=OFF -GNinja ..
```

### Actual behavior ###

```
#14 11.74 [107/110] Linking CXX static library src/libccache_framework.a                                                                                                                              
#14 15.84 [108/110] Linking CXX executable src/test-lockfile                                                                                                                                          
#14 40.64 [109/110] Linking CXX executable ccache                                                                                                                                                     
#14 40.64 FAILED: ccache                                                                                                                                                                              
#14 40.64 : && /usr/bin/g++-8 -fdebug-prefix-map='/tmp/scratch'='/usr/local/src' -g -O3 -DNDEBUG -flto -fno-fat-lto-objects  CMakeFiles/ccache.dir/src/main.cpp.o -o ccache  -Wl,-rpath,/usr/local/lib
:  src/libccache_framework.a  /usr/local/lib/libzstd.so  -pthread  src/third_party/libthird_party.a  src/third_party/blake3/libblake3.a && :                                                          
#14 40.64 /usr/bin/ld: Dwarf Error: Invalid abstract instance DIE ref.                                                                                                                                
#14 40.64 /tmp/ccDPVuoC.ltrans0.ltrans.o: In function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Util::make_path<char const*, char [7]>(char const* const&, char
 const (&) [7]) [clone .constprop.166]':                                                                                                                                                              
#14 40.64 <artificial>:(.text+0xbab1): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'                                                                                      
#14 40.64 <artificial>:(.text+0xbae7): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'                                                                                      
#14 40.64 <artificial>:(.text+0xbb35): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'                                                                            
#14 40.64 <artificial>:(.text+0xbb7f): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'                                                                            
#14 40.64 <artificial>:(.text+0xbb8f): undefined reference to `std::filesystem::__cxx11::path::has_filename() const'                                                                                  
#14 40.64 <artificial>:(.text+0xbbc5): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'                                                                                      
#14 40.64 <artificial>:(.text+0xbbdc): undefined reference to `std::filesystem::__cxx11::path::lexically_normal() const'                                                                              
#14 40.64 <artificial>:(.text+0xbc94): undefined reference to `std::filesystem::__cxx11::path::has_filename() const'                                                                                  
#14 40.64 <artificial>:(.text+0xbcc4): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'                                                                                      
#14 40.64 /usr/bin/ld: Dwarf Error: Invalid abstract instance DIE ref.                                                                                                                                
#14 40.64 /tmp/ccDPVuoC.ltrans9.ltrans.o: In function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Util::make_path<std::__cxx11::basic_string<char, std::char_trai
ts<char>, std::allocator<char> >, char [8]>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char const (&) [8])':
#14 40.64 <artificial>:(.text+0x6f91): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 <artificial>:(.text+0x6fdf): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 <artificial>:(.text+0x7042): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'
#14 40.64 <artificial>:(.text+0x7089): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'
#14 40.64 <artificial>:(.text+0x7099): undefined reference to `std::filesystem::__cxx11::path::has_filename() const'
#14 40.64 <artificial>:(.text+0x70cf): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 <artificial>:(.text+0x70e0): undefined reference to `std::filesystem::__cxx11::path::lexically_normal() const'
#14 40.64 <artificial>:(.text+0x7194): undefined reference to `std::filesystem::__cxx11::path::has_filename() const'
#14 40.64 <artificial>:(.text+0x71ca): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 /usr/bin/ld: Dwarf Error: Invalid abstract instance DIE ref.                                                                                                                                
#14 40.64 /tmp/ccDPVuoC.ltrans10.ltrans.o: In function `Config::read()':                                                                                                                              
#14 40.64 <artificial>:(.text+0x7b8c): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 <artificial>:(.text+0x7be7): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'
#14 40.64 <artificial>:(.text+0x7bf7): undefined reference to `std::filesystem::__cxx11::path::has_filename() const'
#14 40.64 <artificial>:(.text+0x7c2d): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
#14 40.64 <artificial>:(.text+0x7c40): undefined reference to `std::filesystem::__cxx11::path::lexically_normal() const'
#14 40.64 collect2: error: ld returned 1 exit status                                               
#14 57.71 [110/110] Linking CXX executable unittest/unittest
```

### Expected behavior ###

4.6 from a few days ago works just fine.

### Environment ###

Ubuntu 18.04
gcc 8
ccache version 4.7

**Comment by jrosdahl** [comment]
Thanks for the bug report.

However, this is not a bug. :slightly_smiling_face: As mentioned in the [release notes for version 4.7](https://ccache.dev/releasenotes.html#_ccache_4_7), a C++17-capable compiler is now required. For GCC, that means [GCC 9+](https://ccache.dev/platform-compiler-language-support.html#build_time) since GCC 8 doesn't fully support C++17.

That said, I'll convert this issue into an improvement request to support compiling with GCC 8 on Ubuntu 18.04 with the `-lstdc++fs` hack.

For reference, to use ccache on Ubuntu 18.04 already now you can either use the [Linux x86_64 binary release](https://ccache.dev/download.html) or install a newer GCC (for instance from the [ubuntu-toolchain-r/test PPA](https://github.com/ccache/ccache/blob/b53afeb556d55576ed3683fef6566361464393d5/dockerfiles/ubuntu-18.04/Dockerfile#L22-L23)).

**Comment by xkszltl** [comment]
I'm fine with C++17, but if it only take an -lstdc++fs I would say it doesn't worth breaking the compatibility, especially when popular distro like Ubuntu 1804 haven't reached its EOL.

Also, it would be great if compatibility issue can be explicitly listed out.
For example, a simple check in cmake for feature testing, plus a clear error msg within cmake, IMHO is much better than linking error at the end of build, because it's hard to know whether it's expected or not.

**Comment by jrosdahl** [comment]
It has already been implemented, see #1190.
